development/extensions/new_in_rhea_3.3.rst

```diff
@@ -0,0 +1,128 @@
+
+Extension Installation Error Messages
+=====================================
+
+One often requested feature by extension authors finally makes its debut in phpBB 3.3: Displaying error messages to users when an extension can not be enabled!
+
+Typically extension authors use their extension's ``ext.php`` file to set conditional tests to check and see if a phpBB board meets the basic requirements of their extension. If it fails, the extension is not be enabled. However users are only met with an error message that they failed to meet the extension's requirements.
+
+Now extension authors can finally explain what the specific requirements are that caused the extension to fail to install.
+
+This can still be done in the same ``ext.php`` file and the same ``is_enableable()`` method as before. Except now, instead of only being able to return either a true or false boolean, the method allows you to return an array of error messages for each reason why an extension can not be enabled/installed; such as in this following example:
+
+.. code-block:: php
+
+    public function is_enableable()
+    {
+        // Only install extension on phpBB 3.3.0 or newer
+        $enableable = phpbb_version_compare(PHPBB_VERSION, '3.3.0', '>=');
+
+        // If the test failed, return an error message explaining why
+        if (!$enableable)
+        {
+            return array('phpBB 3.3.0 or newer is required to enable this extension.');
+        }
+
+        // Return the result of the test, which would be true if we reached this
+        return $enableable;
+    }
+
+The previous example only needs a slight modification to be backwards compatible with phpBB 3.2 and 3.1. In the following example we can check if the board is phpBB 3.3 and if it is, we can use the new message system, otherwise for older phpBB boards we can use the default system returning a simple true/false boolean:
+
+.. code-block:: php
+
+    public function is_enableable()
+    {
+        // Only install extension on phpBB 3.2.0 or newer
+        $enableable = phpbb_version_compare(PHPBB_VERSION, '3.2.0', '>=');
+
+        // If the test failed and phpBB 3.3 is detected, return error message explaining why
+        if (!$enableable && phpbb_version_compare(PHPBB_VERSION, '3.3.0', '>='))
+        {
+            return array('phpBB 3.2.0 or newer is required to enable this extension.');
+        }
+
+        // Return the result of the test, which could be true (or false for phpBB 3.2 and 3.1)
+        return $enableable;
+    }
+
+Also note that in this one case, we are hard-coding language in the extension. Technically, if your extension could not be enabled, then its language files will not be loaded in the system so using language keys here will not work. There are workarounds to this, however in the interest of cleaner and more elegant code an exception can be made here for allowing hard-coded language strings.
```

Comment:
Should we mention that they should use English in that hard-coding case?

Reply:
tbh, I disagree with the exception, we always have required stuff to be translated, except when the language absolutly wasn't being able to be loaded. Cleaner code is not a reason to hardcode any language IMO

Reply:
Then I will remove the text about cleaner code.